### PR TITLE
[AMORO-2200] Optimize the docker script for amoro

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

fix:#2200.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docker/quickdemo/Dockerfile [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #        .
 ARG AMORO_TAG=master-snapshot
 
-FROM eclipse-temurin:8-jdk-jammy as builder
+FROM alpine as builder
```
Comment (baiyangtx) [inline docker/quickdemo/Dockerfile:28]: There is a conflict between alpine and Flink, which causes Flink to fail to start the task manager.

**Review by baiyangtx — COMMENTED** [review]
I noticed that this PR has done more than one thing. 
1. Modified the Docker image. 
2. Implemented the Spark optimizer. 
These two things are relatively independent, please submit separate PRs to complete them, which is important for maintaining a clear commit history.

**Comment by baiyangtx** [comment]
I noticed that this PR provides a Spark optimizer job but does not provide the implementation related to the Spark optimizer container. 
Can you implement this part of the code? Otherwise, we can only use the Spark optimizer in an external container.

AND, 

You can refer to the Flink optimizer module, and the Spark optimizer module should be implemented in a separate module. This way, the project can be compiled more flexibly.

**Comment by tcodehuber** [comment]
> I noticed that this PR has done more than one thing.
> 
> 1. Modified the Docker image.
> 2. Implemented the Spark optimizer.
> 
> These two things are relatively independent, please submit separate PRs to complete them, which is important for maintaining a clear commit history.

sorry, my fault. Now i have correct it.

**Review by baiyangtx — COMMENTED** [review]
Have you tested the quickdemo process? As far as I know, there is a conflict between the Alpine image and the Flink engine, which prevents the Flink task from starting.

**Comment by tcodehuber** [comment]
> Have you tested the quickdemo process? As far as I know, there is a conflict between the Alpine image and the Flink engine, which prevents the Flink task from starting.

Not yet. How can i close this PR?

**Comment by baiyangtx** [comment]
> > Have you tested the quickdemo process? As far as I know, there is a conflict between the Alpine image and the Flink engine, which prevents the Flink task from starting.
> 
> Not yet. How can i close this PR?

I will close it.

**Comment by baiyangtx** [comment]
The Spark optimizer is useful. Are you willing to submit a new Pull Request (PR)?  @tcodehuber

**Comment by tcodehuber** [comment]
> The Spark optimizer is useful. Are you willing to submit a new Pull Request (PR)? @tcodehuber

OK, I will try it later.

**Pull request closed without merging**